app/api/routes/product.py

```diff
@@ -1,138 +1,112 @@
-
-    try:
-        validator.validate_unique_name(product_data.name)
-        validator.validate_product_data(product_data)
-
-        new_product = Product(**product_data.dict())
-
-        db.add(new_product)
-        db.commit()
-        db.refresh(new_product)
+    service = ProductService(db)
 
-        return new_product
-    except InvalidProductDataException as e:
-        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=e.detail)
-    except Exception as e:
+    if service._is_product_existing(name=product_data.name):
         raise HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e)
+            status_code=status.HTTP_409_CONFLICT, detail="Product already exists."
         )
 
+    new_product = Product(**product_data.dict())
+    db.add(new_product)
+    db.commit()
+    db.refresh(new_product)
 
-@router.get("/products/{product_id}", response_model=ProductResponse)
-def get_product(product_id: UUID, db: Session = Depends(get_db)):
-    service = ProductService(db)
-    try:
-        return service.get_product_by_id(product_id)
-    except ProductNotFoundException as e:
-        raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail=str(e),
-        )
+    return new_product
 
 
-@router.put("/products/{product_id}", response_model=ProductResponse)
-def update_product(
-    product_id: UUID, product_data: ProductUpdate, db: Session = Depends(get_db)
+@router.get("/{product_id}", response_model=ProductResponse)
+def get_product(
+    product_id: str,
+    db: Session = Depends(get_db),
 ):
-    validator = ProductValidator(db)
-    service = ProductService(db)
 
     try:
-        existing_product = service.get_product_by_id(product_id)
-        if not existing_product:
-            raise ProductNotFoundException(product_id)
-
-        if product_data.name and product_data.name != existing_product.name:
-            validator.validate_unique_name(product_data.name)
-
-        validator.validate_product_data(product_data)
-
-        updated_product = service.update_product(product_id, product_data)
-
-        return updated_product
-    except ProductNotFoundException as e:
+        product_id = UUID(product_id)
+    except ValueError:
         raise HTTPException(
-            status_code=status.HTTP_404_NOT_FOUND,
-            detail=str(e),
+            status_code=status.HTTP_400_BAD_REQUEST, detail="Invalid UUID!"
         )
-    except InvalidProductDataException as e:
-        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=e.detail)
-    except Exception as e:
+
+    service = ProductService(db)
+    try:
+        return service.get_product_by_id(product_id)
+    except ProductNotFoundException:
         raise HTTPException(
-            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
-            detail=str(e),
+            status_code=status.HTTP_404_NOT_FOUND, detail="Product not found."
         )
 
```

Comment:
Please add the general exception

Reply:
done a whole general exception handler for every routes if any unexpected exception.